src/agora/txrx/workers/txrx_worker_argos.cc

```diff
@@ -0,0 +1,730 @@
+  program_start_ticks_ = GetTime::Rdtsc();
+  while (Configuration()->Running() == true) {
+    if (0 == DequeueSend(time0)) {
+      // attempt to receive symbol (might be good to reduce the rx timeout to allow for tx during dead time)
+      if (kSymbolTimingEnabled) {
+        rx_times.at(receive_attempt.interface_).start_ticks_ = GetTime::Rdtsc();
+      }
+      auto pkts = RecvEnqueue(receive_attempt.interface_, global_frame_id,
+                              global_symbol_id);
+
+      const size_t rx_time_ticks = GetTime::Rdtsc();
+      if (kSymbolTimingEnabled) {
+        rx_times.at(receive_attempt.interface_).end_ticks_ = rx_time_ticks;
+      }
+
+      if (!pkts.empty()) {
+        RtAssert(pkts.size() == channels_per_interface_,
+                 "Received data but it was the wrong dimension");
+        size_t rx_symbol_id = pkts.front()->symbol_id_;
+
+        if (unlikely(rx_symbol_id != receive_attempt.symbol_)) {
+          MLPD_ERROR(
+              "TxRxWorkerArgos[%zu]: Frame %d - Expected symbol %zu but "
+              "received %zu\n",
+              tid_, pkts.front()->frame_id_, receive_attempt.symbol_,
+              rx_symbol_id);
+        }
+
+        RtAssert(
+            rx_symbol_id == receive_attempt.symbol_,
+            "The expected receive symbol does not match the actual symbol");
+
+        // Symbol received, change the rx interface
+        TxRxWorkerRx::RxParameters successful_receive = receive_attempt;
+        receive_attempt = UpdateRxInterface(successful_receive);
+        MLPD_TRACE(
+            "TxRxWorkerArgos[%zu]: Last Interface %zu - symbol %zu:%zu, next "
+            "interface %zu - symbol %zu\n",
+            tid_, successful_receive.interface_, successful_receive.symbol_,
+            rx_symbol_id, receive_attempt.interface_, receive_attempt.symbol_);
+
+        if (kIsWorkerTimingEnabled) {
+          const auto frame_id = pkts.front()->frame_id_;
+          if (frame_id > prev_frame_id) {
+            rx_frame_start_[frame_id % kNumStatsFrames] = rx_time_ticks;
+            prev_frame_id = frame_id;
+          }
+        }
+
+        PrintRxSymbolTiming(rx_times, pkts.front()->frame_id_,
+                            successful_receive.symbol_,
+                            receive_attempt.symbol_);
+      }  // if (pkt.size() > 0)
+    }    // DequeueSendArgos(time0) == 0
+  }      // Configuration()->Running() == true
+  running_ = false;
+}
+
+//RX data, should return channel number of packets || 0
+std::vector<Packet*> TxRxWorkerArgos::RecvEnqueue(size_t interface_id,
+                                                  size_t global_frame_id,
+                                                  size_t global_symbol_id) {
+  const size_t radio_id = interface_id + interface_offset_;
+  const size_t ant_id = radio_id * channels_per_interface_;
+  const size_t cell_id = Configuration()->CellId().at(radio_id);
+
+  std::vector<size_t> ant_ids(channels_per_interface_);
```

Comment:
This vector seems redundant now. We can just use channels_per_interface_ to iterate over antennas per radio.

Reply:
Will fix in next PR